 Fix a NPE when putting a null-Bundle in an Intent
Happens frequently e.g. with GoogleAccountAuthenticator

diff --git a/core/java/android/content/Intent.java b/core/java/android/content/Intent.java
index 0262f54..f9b5de0 100644
--- a/core/java/android/content/Intent.java
+++ b/core/java/android/content/Intent.java
@@ -7790,6 +7790,9 @@ public Intent putExtras(Intent src) {
      * @see #removeExtra
      */
     public Intent putExtras(Bundle extras) {
+        if (extras == null) {
+            return this;
+        }
         if (mExtras == null) {
             mExtras = new Bundle();
         }
